config/iam/config.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package iam
+
+import (
+	"github.com/crossplane-contrib/terrajet/pkg/config"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+var SelfPgkPath = "github.com/crossplane-contrib/provider-tf-aws/config/iam"
+
+func ExternalNameAsName(base map[string]interface{}, externalName string) {
+	base["name"] = externalName
+}
+
+func PolicyARNExtractor() reference.ExtractValueFn {
+	return func(mg resource.Managed) string {
+		paved, err := fieldpath.PaveObject(mg)
```

Comment:
This is a high cost operation and I think we can avoid it with simple [type casting](https://github.com/crossplane/provider-aws/blob/c78bf9a/apis/identity/v1beta1/referencers.go#L32).

Reply:
As we discussed offline, type casting causes some compilation issues which makes code generation a multi-step manual process. I believe the operation cost here could be ignored considering this is usually a one-time operation and the effect of this on our performance bottleneck should be negligible

Reply:
FTR, this what I am getting when I try strong type: 

```
package command-line-arguments
	imports github.com/crossplane-contrib/provider-tf-aws/apis
	imports github.com/crossplane-contrib/provider-tf-aws/apis/iam/v1alpha1
	imports github.com/crossplane-contrib/provider-tf-aws/config/iam
	imports github.com/crossplane-contrib/provider-tf-aws/apis/iam/v1alpha1: import cycle not allowed
```